pallets/blindbox/src/lib.rs

```diff
@@ -0,0 +1,288 @@
+    pub type BlindBoxes<T: Config> = StorageMap<_, Twox64Concat, BlindBoxId, (), OptionQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn get_blindboxescreators)]
+    pub type BlindBoxesCreators<T: Config> = StorageMap<_, Twox64Concat, T::AccountId, (), OptionQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn all_blindboxes_count)]
+    pub(super) type AvailableBlindBoxesCount<T: Config> = StorageValue<_, u32, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn get_available_ksm)]
+    pub(super) type AvailableKSM<T: Config> = StorageValue<_, u32, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn is_init)]
+    pub(super) type Init<T: Config> = StorageValue<_, bool, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn nonce)]
+    pub(super) type Nonce<T: Config> = StorageValue<_, u32, ValueQuery>;
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub (super) fn deposit_event)]
+    #[pallet::metadata(T::AccountId = "AccountId")]
+    pub enum Event<T: Config> {
+        RandomnessConsumed(H256, H256),
+        BlindBoxIdGenerated(Vec<u32>),
+        BlindBoxOpened(T::AccountId, BlindBoxType, u32),
+        BlindBoxGoodLuckNextTime(u32),
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        //No permission
+        NoPermission,
+        // BlindBox does not exist
+        BlindBoxDoesNotExist,
+        // BlindBoxes still available, only allow to create once all the blindboxes have been used
+        BlindBoxesStillAvailable
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[pallet::weight(10_000)]
+        pub(super) fn set_blindbox_caller(origin: OriginFor<T>, account_id: T::AccountId ) -> DispatchResultWithPostInfo {
+            let _ = ensure_root(origin)?;
+
+            BlindBoxesCreators::<T>::insert(account_id, {});
+
+            Ok(().into())
+        }
+
+        #[pallet::weight(10_000)]
+        pub(super) fn generate_blindbox_ids(origin: OriginFor<T>, number_blindboxes: u32) -> DispatchResultWithPostInfo {
+            let caller = ensure_signed(origin)?;
+
+            // Ensure the authorized caller can call this func
+            ensure!(
+                BlindBoxesCreators::<T>::contains_key(&caller),
+                Error::<T>::NoPermission
+            );
+
+            // Ensure caller can only generate blindboxes once all the available blindboxes have been used
+            ensure!(
+                AvailableBlindBoxesCount::<T>::get() <= 0,
+                Error::<T>::BlindBoxesStillAvailable
+            );
+
+            let mut blindbox_vec = Vec::new();
+
+            // Generate random blindbox id and store
+            let mut number_blindboxes_generated = 0;
+            let mut i = 0;
+
+            // Add safe check in case of infinite loop, running extra 10 loops to generate unique blindbox id
+            while number_blindboxes_generated < number_blindboxes && i < number_blindboxes + 10 {
```

Comment:
This should check against a MAX type e.g MaxNumberOfBlindBox to ensure while loop never runs infinitive or halt blockchain. For instance, number_blindboxes is u32 and user input is u32::MAX, when you add 10, the blockchain will panic.

Reply:
Similar approach with MaxGenerateRandom